utils/utils.jsx

```diff
@@ -1776,6 +1805,68 @@ export function handleFormattedTextClick(e, currentRelativeTeamUrl) {
         if (!(e.button === MIDDLE_MOUSE_BUTTON || e.altKey || e.ctrlKey || e.metaKey || e.shiftKey)) {
             e.preventDefault();
 
+            const state = store.getState();
+            const user = getCurrentUser(state);
+            if (isSystemAdmin(user.roles)) {
+                const match = isChannelOrPermalink(linkAttribute.value);
+                if (match) {
```

Comment:
There are a lot of nested if statements inside here and I wonder if we could reduce how many we have. It could lead to complications in the future when someone has to make updates. We can remove the first if statement and do this?:
```suggestion
                if (isSystemAdmin(user.roles) && match) {
```

Anywhere we have something like this:
```
const {data: channelData} = await store.dispatch(getChannelAction(post.channel_id));
if (channelData) {
      channel = channelData;
}
```

We could change to this:
```
const {data: channelData} = await store.dispatch(getChannelAction(post.channel_id));
channel = channelData || channel;
```

These are just suggestions but it might make it more readable. There are probably other places we can clean up in this block too

Reply:
Agreed and I refactored code a bit now.

Reply:
Nice, thank you!